### ISSUE TITLE
a11y: Fix keyboard navigation, role, aria-labels and focus display issues on the side navigation component

### ISSUE THREAD

**Issue by nwmac** [rancher/dashboard#12780]
a11y Issues: 35, 40, 41

**Issue 35**
The side navigation expand/collapse button (see below) does not show keyboard focus and does not response to enter/space to activate.

![image](https://github.com/user-attachments/assets/2f797991-8b68-47f8-ae4b-efdd4e8dbc5c)

**Issue 40**

A focus indicator is not visually available as user tabs through mentioned interactive elements using a keyboard resulting in users with learning and mobility impairments as well as those with low vision not being able to easily find out which element currently has the focus.

Header Area 
- Import YAML 
- Kubectl Shell
- Download KubeConfig
- Copy KubeConfig to Clipboard
- Resource Search 
- Projects/Namespaces
- Nodes 1
- Cluster and Project Members
- Events 0 and so on. 

![image](https://github.com/user-attachments/assets/23aee28c-95b3-4ef2-a979-a4b7183c1eaa)

**Issue 41**

On activating the mentioned instances, the content below gets expanded/collapsed dynamically, although the dynamic change in the content is easier to understand for sighted users, it is not intuitive for screen reader users.

- Main Content 
- Starred 
- Cluster
- Workloads
- Apps
- Service Discovery 
and so on.

![image](https://github.com/user-attachments/assets/88b56011-1b21-46c6-acd5-b7cdb5ae721e)

**Comment by yonasberhe23** [comment]
Needs manual validation.

**Comment by IsaSih** [comment]
Observations:
for the side menu, the space bar does not select the sub-options of the side menu, just the main options. Is this expected?

<img width="327" alt="Image" src="https://github.com/user-attachments/assets/0e6f4664-86a8-4687-9008-1812e2bd00af" />

**Comment by IsaSih** [comment]
As shown in the screenshot, the test fails issue 40. There's no focus nor tab navigation in the header area. Feature is working only for the Projects/Namespaces selection and the user logout icon.

Tested in 
Rancher | v2.11-a1bda936c809e0394520c9d40d2998529223401e-head
-- | --
Dashboard | master c987984
Rancher CLI | v2.11.0-rc.1

**Comment by aalves08** [comment]
> As shown in the screenshot, the test fails issue 40. There's no focus nor tab navigation in the header area. Feature is working only for the Projects/Namespaces selection and the user logout icon.
> 
> Tested in
> 
> Rancher	v2.11-a1bda936c809e0394520c9d40d2998529223401e-head
> Dashboard	master c987984
> Rancher CLI	v2.11.0-rc.1

I forgot that this had header items associated with the issue. Please ignore those 🙏 

We've decided to tackle the header part in one go, since it's a problematic area to fix in terms of accessibility, especially because of the modals part. I would say that the following issues track the work needed on the header:

https://github.com/rancher/dashboard/issues/12790
https://github.com/rancher/dashboard/issues/12772
https://github.com/rancher/dashboard/issues/8113
https://github.com/rancher/dashboard/issues/12779
https://github.com/rancher/dashboard/issues/12778

I've got a WIP PR for the header: https://github.com/rancher/dashboard/pull/13011/, which I am tackling at the moment.

So I would say to ignore the header part on this issue for now.

FYI @rak-phillip

**Comment by edenhernandez-suse** [comment]
> Also, I observed that by selecting a cluster in the burger menu bar, the tab navigation doesnt change the focus to the side navigation component, but stays in the burger menu bar. In terms of usability, I suggest that once a cluster is selected from the burger menu, that the focus should then move to the side navigation so that the user accesses the cluster resources

This is a tricky one in our environment since our title (the cluster name) is in the top bar and not inside the main content. My approach to this one would be to shift focus to the side navigation, with a first hidden tab element for "Skip to content" before the first navigation item (in case of doubts, I'm talking about this: [Skip to main content](https://www.a11y-collective.com/blog/skip-to-main-content/)). That would allow users to directly jump to the cluster dashboard without tabbing through the entire side navigation.

**Comment by aalves08** [comment]
Waiting for feedback on this issue 🙏
